runtime/filetype.vim

```diff
@@ -680,12 +681,8 @@ if !empty($XDG_CONFIG_HOME)
 endif
 au BufNewFile,BufRead git-rebase-todo		setf gitrebase
 au BufRead,BufNewFile .gitsendemail.msg.??????	setf gitsendemail
-au BufNewFile,BufRead .msg.[0-9]*
-      \ if getline(1) =~ '^From.*# This line is ignored.$' |
-      \   setf gitsendemail |
-      \ endif
 au BufNewFile,BufRead *.git/*
-      \ if getline(1) =~ '^\x\{40\}\>\|^ref: ' |
+      \ if getline(1) =~# '^\x\{40,\}\>\|^ref: ' |
       \   setf git |
       \ endif
```

Comment:
Do you want to include a port of this into filetype.lua?

I think this ought to do it?

```lua
[".*%.git/.*"] = function(path, bufnr)
  local firstline = getline(1)
  if firstline:find("^" .. string.rep("%x", 40) .. "+ ") or firstline:sub(1, 5) == "ref: " then
    return "git"
  end
end,
```

Reply:
Worth a try -- I trust you on the Lua regex! What about the test?

Reply:
Or maybe better to include it in https://github.com/neovim/neovim/pull/16959#pullrequestreview-845786399 (which also needs a similar test).

Reply:
I'm merging this now while ASAN is still green, so the `git` pattern should go into the linked PR.